tests/common/test_logger.py

```diff
@@ -229,6 +229,98 @@ def test_telemetry_logger_not_on_by_default(self, mock_save):
+
+        counter = 0
+        for appender in lg.appenders:
+            if isinstance(appender, logger.FileAppender):
+                counter += 1
+            elif isinstance(appender, logger.TelemetryAppender):
+                counter += 1
+            elif isinstance(appender, logger.ConsoleAppender):
+                counter += 1
+            elif isinstance(appender, logger.StdoutAppender):
+                counter += 1
+
+        # All 4 appenders should have been included.
+        self.assertEquals(4, counter)
+
+        lg.warn("Test Log")
+        self.assertEqual(mock_file_write.call_count, 1)
+        self.assertEqual(mock_console_write.call_count, 1)
+        self.assertEqual(mock_telem_write.call_count, 1)
+        self.assertEqual(mock_stdout_write.call_count, 1)
+
+        lg.info("Test Log")
+        self.assertEqual(mock_file_write.call_count, 2)
+        self.assertEqual(mock_console_write.call_count, 2)
+        self.assertEqual(mock_telem_write.call_count, 2)
+        self.assertEqual(mock_stdout_write.call_count, 2)
+
+        lg.error("Test Log")
+        self.assertEqual(mock_file_write.call_count, 3)
+        self.assertEqual(mock_console_write.call_count, 3)
+        self.assertEqual(mock_telem_write.call_count, 3)
+        self.assertEqual(mock_stdout_write.call_count, 3)
+
+    @patch("azurelinuxagent.common.logger.StdoutAppender.write")
+    @patch("azurelinuxagent.common.logger.TelemetryAppender.write")
+    @patch("azurelinuxagent.common.logger.ConsoleAppender.write")
+    @patch("azurelinuxagent.common.logger.FileAppender.write")
+    def test_set_prefix(self, mock_file_write, mock_console_write, mock_telem_write, mock_stdout_write):
+        lg = logger.Logger(logger.DEFAULT_LOGGER)
+        prefix = "YoloLogger"
+
+        lg.set_prefix(prefix)
+        self.assertEquals(lg.prefix, prefix)
+
+        lg.add_appender(logger.AppenderType.FILE, logger.LogLevel.INFO, path=self.log_file)
+        lg.add_appender(logger.AppenderType.TELEMETRY, logger.LogLevel.WARNING, path=add_log_event)
+        lg.add_appender(logger.AppenderType.CONSOLE, logger.LogLevel.WARNING, path="/dev/null")
+        lg.add_appender(logger.AppenderType.STDOUT, logger.LogLevel.WARNING, path=None)
+
+        lg.error("Test Log")
+
+        self.assertIn(prefix, mock_file_write.call_args[0][1])
+        self.assertIn(prefix, mock_console_write.call_args[0][1])
+        self.assertIn(prefix, mock_telem_write.call_args[0][1])
+        self.assertIn(prefix, mock_stdout_write.call_args[0][1])
+
+    @patch("azurelinuxagent.common.logger.StdoutAppender.write")
+    @patch("azurelinuxagent.common.logger.TelemetryAppender.write")
+    @patch("azurelinuxagent.common.logger.ConsoleAppender.write")
+    @patch("azurelinuxagent.common.logger.FileAppender.write")
+    def test_nested_logger(self, mock_file_write, mock_console_write, mock_telem_write, mock_stdout_write):
```

Comment:
I didnt quite get how this test works, can you explain a bit please?

Reply:
+1

Reply:
The purpose of this test is to see if the logger gets correctly created when passed it another logger. And also if the appender correctly gets the messages This is how the `ExtHandlerInstance` logger works. There was no test to see this.

What I do is initialize the default logger (logger.add_logger_appender(*)). Then create a new logger from it
```
lg = logger.Logger(logger.DEFAULT_LOGGER, child_prefix)
```

and then log using `lg`, and `logger`. And see if both logs are flowing through or not. This already works as we can see `ExtHandlerInstance` logs and `DEFAULT_LOGGER` logs are available.